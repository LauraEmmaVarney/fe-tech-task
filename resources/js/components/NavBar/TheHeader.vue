<template>
    <header 
        role="banner"
        class="container header-checkout my-5"
    >
        <div class="row mb-2">
            <div 
                v-once
                class="col-12 col-xl-3 my-xl-auto text-center text-xl-left"
            >
                <img 
                    v-if="store.logoUrl" 
                    :src="store.logoUrl" 
                    class="animated fadeInDown faster rounded" 
                    style="max-width:100%; max-height:100px;"
                />
                <h3 
                    v-else 
                    class="mb-0 animated fadeInDown faster"
                >
                    {{ store.name }}
                </h3>
            </div>
            <div class="col-12 col-md-12 col-lg-10 col-xl-8 ml-xl-auto mx-xl-0 mx-auto my-auto px-0 pt-5 pt-xl-0 d-none d-md-block">
                <div class="row">
                    <NavBarItem 
                        :step="1"
                        :active-step="activeStep"
                        label="Your Details"
                        icon="fa-user-circle"
                    />
                    <NavBarItem 
                        :step="2"
                        :active-step="activeStep"
                        label="Card Details"
                        icon="fa-credit-card"
                    />
                    <NavBarItem 
                        :step="3"
                        :active-step="activeStep"
                        label="Success!"
                        icon="fa-check"
                    />
                </div>
            </div>
        </div>
    </header>
</template>

<script>

    import NavBarItem from "./NavBarItem.vue";

    export default {
        name: "HeaderWrapper",
        components: {
            NavBarItem
        },
        props: {
            store: {
                type: Object,
                required: true
            },
            activeStep: {
                type: Number,
                required: true,
            }
        }
    }

</script>