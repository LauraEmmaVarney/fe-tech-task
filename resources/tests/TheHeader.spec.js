import { shallowMount } from '@vue/test-utils'
import TheHeader from '../js/components/NavBar/TheHeader.vue';

describe('TheHeader component tests', () => {
    const wrapper = shallowMount(TheHeader, {
        propsData: {
            activeStep: 1,
            store: {
                "currencySymbol":"$",
                "name":"Test Store",
                "logoUrl": null
            },
        }
    });

    it('mounts the component', () => {
        expect(wrapper.exists()).toBe(true);
    });

    it('checks props data', () => {
        expect(wrapper.props().activeStep).toBe(1);
        expect(wrapper.props().store).toEqual({
            "currencySymbol":"$",
            "name":"Test Store",
            "logoUrl": null
        });
    });

    it('check store name', () => {
        const title = wrapper.find('h3');
        expect(title.text()).toBe('Test Store');
    });

    it('check against jest snapshot test', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
})