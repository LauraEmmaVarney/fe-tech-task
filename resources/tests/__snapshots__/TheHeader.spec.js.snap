// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheHeader component tests check against jest snapshot test 1`] = `
<header
  class="container header-checkout my-5"
  role="banner"
>
  <div
    class="row mb-2"
  >
    <div
      class="col-12 col-xl-3 my-xl-auto text-center text-xl-left"
    >
      <h3
        class="mb-0 animated fadeInDown faster"
      >
        
                Test Store
            
      </h3>
    </div>
     
    <div
      class="col-12 col-md-12 col-lg-10 col-xl-8 ml-xl-auto mx-xl-0 mx-auto my-auto px-0 pt-5 pt-xl-0 d-none d-md-block"
    >
      <div
        class="row"
      >
        <navbaritem-stub
          activestep="1"
          icon="fa-user-circle"
          label="Your Details"
          step="1"
        />
         
        <navbaritem-stub
          activestep="1"
          icon="fa-credit-card"
          label="Card Details"
          step="2"
        />
         
        <navbaritem-stub
          activestep="1"
          icon="fa-check"
          label="Success!"
          step="3"
        />
      </div>
    </div>
  </div>
</header>
`;
